lib/elixir/lib/calendar/duration.ex

```diff
@@ -197,4 +197,144 @@ defmodule Duration do
       microsecond: {-ms, p}
     }
   end
+
+  @doc """
+  Parses an ISO 8601-2 formatted duration string to a `Duration` struct.
+
+  ## Examples
+
+      iex> Duration.from_iso8601("P1Y2M3DT4H5M6S")
+      {:ok, %Duration{year: 1, month: 2, day: 3, hour: 4, minute: 5, second: 6}}
+      iex> Duration.from_iso8601("PT10H30M")
+      {:ok, %Duration{hour: 10, minute: 30, second: 0}}
+      iex> Duration.from_iso8601("P3Y-2MT3H")
+      {:ok, %Duration{year: 3, month: -2, hour: 3}}
+      iex> Duration.from_iso8601("-P3Y2MT3H")
+      {:ok, %Duration{year: -3, month: -2, hour: -3}}
+      iex> Duration.from_iso8601("-P3Y-2MT3H")
+      {:ok, %Duration{year: -3, month: 2, hour: -3}}
+
+  """
+  @spec from_iso8601(String.t()) :: {:ok, t} | {:error, String.t()}
+  def from_iso8601("P" <> duration_string) do
+    parse(duration_string, %{}, "", false)
+  end
+
+  def from_iso8601("-P" <> duration_string) do
```

Comment:
Is this supported by the spec? Or can we skip it for now?

Reply:
That's part of the ISO8601 spec extension - unfortunately i can't find much about it but it's mentioned here for example: https://tc39.es/proposal-temporal/docs/duration.html

> As extensions to the standard, ISO 8601-2 allows a sign character at the start of the string, and allows combining weeks with other units. If you intend to use a string such as P3W1D, +P1M, or -P1M for interoperability, note that other programs may not accept it.

It's supported by some but not all libraries i took a look at